WeakAuras/GenericTrigger.lua

```diff
@@ -1337,29 +1330,18 @@ function GenericTrigger.Add(data, region)
                     hasParam = true
                   end
                 elseif trueEvent:match("^UNIT_") then
-                  local includePets
-                  if i == "grouppets" then
+                  isUnitEvent = true
+
+                  if string.lower(strsub(i, #i - 3)) == "pets" then
+                    i = strsub(i, 1, #i-4)
                     includePets = "PlayersAndPets"
-                    i = "group"
-                  elseif i == "grouppetsonly" then
+                  elseif string.lower(strsub(i, #i - 7)) == "petsonly" then
                     includePets = "PetsOnly"
-                    i = "group"
-                  end
-                  MultiUnitLoop(
-                    function(unit)
-                      trigger_unit_events[unit] = trigger_unit_events[unit] or {}
-                      tinsert(trigger_unit_events[unit], trueEvent)
-                      isUnitEvent = true
-                    end, i, includePets
-                  )
-                  if includePets ~= nil then
-                    MultiUnitLoop(
-                      function(unit)
-                        trigger_unit_events[unit] = trigger_unit_events[unit] or {}
-                        tinsert(trigger_unit_events[unit], "UNIT_PET")
```

Comment:
I did remove the special handling of UNIT_PET here. Custom coders will need to add UNIT_PET:group to their event list if they want that event. That sounds fine to me.

For internal triggers, we listen to our fake UNIT_CHANGED_* events, so that work without any problem